### DISCUSSION TITLE
Passing `scoped_session` method dependency to path dependency triggers `Invalid args for response field` pydantic error

### DISCUSSION THREAD

**Discussion by NoahY43619** [fastapi/fastapi#6143] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Any

from fastapi import APIRouter
from fastapi import Depends
from pydantic import BaseModel
from sqlalchemy import create_engine
from sqlalchemy.orm import scoped_session
from sqlalchemy.orm import sessionmaker

DB_URL = "some_db_url"

class TestSuccess(BaseModel):
    message: str = "Test success"
    code: int = 200
    param_1: str


router = APIRouter(prefix="/test_router")

def get_db_session() -> scoped_session:

    engine = create_engine(DB_URL)
    session_maker = sessionmaker(bind=engine)
    db_session = scoped_session(session_maker)

    return db_session

def query_db(param_1: str, db: scoped_session) -> bool

    # return True if param_1 in some db table else raise HTTPException


@router.post(
    path="/test", 
    response_model=TestSuccess,
    dependencies=[Depends(query_db)], # <--------- Removing this dependency solves the error
)
def test_method_1(
    param_1: str,
    db: scoped_session = Depends(get_db_session),
) -> dict[str, Any]:

    # Do stuff with db
    # ...
    
    response = {"param_1": param_1}
    
    return response
```

### Description

Hi everyone,

I've been having some issues with overriding a database dependency in my app for testing purposes, which seem to come from a conflict with the use of the `response_model` path parameter.

As suggested by the docs, I tried to use a dependency returning an SQLAlchemy `scoped_session` as default method parameter, so that I would then be able to override it in my tests. A simplified version of the code is available above. This method fails with message:

```
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that <class 'sqlalchemy.orm.scoping.scoped_session'> is a valid pydantic field type
```

This seems to be due to the fact that the `db` parameter passed from the method dependency to the path dependency is not recognised by pydantic as valid. This error is suppressed by removing the `query_db` path dependency.

Can anyone suggest a correct way of achieving this?

Thanks in advance for the help.

### Operating System

Linux

### Operating System Details

Debian 11

### FastAPI Version

0.85.0

### Python Version

3.9.2

### Additional Context

pydantic==1.10.2

**Comment by JarroVGIT** [comment, marked as answer]
```
class ResponseModel(BaseModel):
    message: str = "Test success"
    code: int = 200
    param_1: str
```
Can we assume that wherever you use `TestSuccess` as response model, you actually mean `ResponseModel`? 

I found the `422` in case two weird but cannot reproduce it. Maybe you don't import `Depends` somehow? The error suggests that you should pass a query parameter named `db_url`, so it doesn't solve for your dependency. Dependencies do not show up in the docs, you can try it with the below MRE (will run as is):
```python
from fastapi import Depends, FastAPI

app = FastAPI()

def get_string():
    return "Marc"

@app.get("/")
async def root(string: str = Depends(get_string)):
    return {"oh hi": string}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)
```
You'll see that `string` will not show up in the docs.

Lastly; you don't have to cast to the response model, then to dict(), and then return that dictionary. You should just return `{"param_1": param1}`, and FastAPI will cast it for you to `TestSuccess`. That is the point of specifying a response_model.

As for Case 1; you might want to inspect how you set param_1. You say it is coming from the Query Param (or you imply that with your code) but I would bet you are changing that parameter with something from the database, which is not a string.
Either way; the two cases show very different problems that are not really related, would be my guess.

> Between Case 1 and Case 2, which would be the most sensible way to go about what I am trying to do, and how can I get it to work?

There is good documentation how to deal with database connections via dependencies in the [documentation here](https://fastapi.tiangolo.com/tutorial/sql-databases/?h=sql#create-a-dependency). I would recommend to follow that approach.

**Comment by NoahY43619** [comment]
Hi @JarroVGIT , and thank you for your answer.

I realised after reading your reply that my question was indeed two-pronged and poorly formulated. I have edited my question and the code sample to make the issue both clearer and more correct.

Also, I went overboard when first attempting to simplify the code sample, and omitted the part that seems to be actually triggering the error: the path dependency.

Apologies for the low-quality question. I hope its second version will make it less turbid.

Thanks again for the help, and please do let me know if there is anything I can do to make the issue clearer.

**Comment by JarroVGIT** [comment]
Could you try to change:
`def query_db(param_1: str, db: scoped_session) -> bool`

into this:
`def query_db(param_1: str,  db: scoped_session = Depends(get_db_session)) -> bool`

I am not sure if FastAPI will pass a parameter from your path operation function to a dependency that is declared in the path decorator.

**Comment by NoahY43619** [comment]
Thanks for the tip @JarroVGIT, it worked instantly.

Closing the issue as everything is running fine now. 

Thanks again four your help.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT  ! 👏  🙇 

Thanks for reporting back and closing the issue @NoahY43619  👍
